doc/modules/ROOT/pages/design/middleware.adoc

```diff
@@ -275,3 +275,40 @@ a middleware descriptor's `:requires` set contains
 * Any requests it handles will contain the key `:nrepl.middleware.print/print-fn`,
   whose value is a function that calls the given printer function with the given
   options – i.e. its signature is `[value writer]`.
+
+== Evaluation Errors
+
+nREPL includes a `caught` middleware which provides a configurable hook for any
+`java.lang.Throwable` that should be conveyed interactively (generally by
+printing to `\*err*`). Like the `print` middleware, any options may be provided
+in either requests or responses (the former taking precedence over the latter if
+any options are specified in both). The following options are supported:
+
+* `:nrepl.middleware.caught/caught`: a fully-qualified symbol naming a var whose
```

Comment:
This should probably be `caught-fn` for consistency with `print-fn`.

Reply:
I didn't want to use the same name for both options – we use `caught` for the name of the var (i.e. only at request level) and `caught-fn` for the actual function object. Using `caught` / `print` for the "var name" semantic in the request is consistent with the [`eval` option](https://github.com/nrepl/nrepl/blob/0.5.3/src/clojure/nrepl/middleware/interruptible_eval.clj#L256) in the eval middleware.